test/test_linalg.py

```diff
@@ -922,6 +923,90 @@ def test_nuclear_norm_exceptions_old(self, device):
 
 
+    @precisionOverride({torch.float: 1e-2, torch.cfloat: 1e-4})
+    @skipCUDAIfNoMagma
+    @skipCPUIfNoLapack
+    @dtypesIfCPU(torch.float32, torch.float64, torch.complex64, torch.complex128)
+    @dtypesIfCUDA(torch.float32, torch.float64)
+    def test_cholesky(self, device, dtype):
+        from torch.testing._internal.common_utils import random_hermitian_pd_matrix
+
+        def run_test(shape, batch):
+            A = random_hermitian_pd_matrix(shape, *batch, dtype=dtype, device=device)
+            expected_L = np.linalg.cholesky(A.cpu().numpy())
+            actual_L = torch.linalg.cholesky(A)
+            self.assertEqual(actual_L, expected_L)
+
+        shapes = (0, 3, 5)
+        batches = ((), (3, ), (2, 2))
+        larger_input_case = [(100, (5, ))]
+        for shape, batch in list(itertools.product(shapes, batches)) + larger_input_case:
+            run_test(shape, batch)
+
+        # cholesky requires the input to be a square matrix
+        A = torch.randn(2, 3, device=device, dtype=dtype)
+        with self.assertRaisesRegex(RuntimeError, r'must be batches of square matrices'):
+            torch.linalg.cholesky(A)
+        with self.assertRaisesRegex(np.linalg.LinAlgError, r'Last 2 dimensions of the array must be square'):
+            np.linalg.cholesky(A.cpu().numpy())
+
+        # cholesky requires the input to be a matrix
+        A = torch.randn(2, device=device, dtype=dtype)
+        with self.assertRaisesRegex(RuntimeError, r'must have at least 2 dimensions'):
+            torch.linalg.cholesky(A)
+        with self.assertRaisesRegex(np.linalg.LinAlgError, r'1-dimensional array given\. Array must be at least two-dimensional'):
+            np.linalg.cholesky(A.cpu().numpy())
+
+        # if the input matrix is singular, an error should be raised
+        A = torch.eye(3, 3, dtype=dtype, device=device)
+        A[-1, -1] = 0  # Now A is singular
+        with self.assertRaisesRegex(RuntimeError, r'U\(3,3\) is zero, singular U\.'):
+            torch.linalg.cholesky(A)
+        with self.assertRaisesRegex(np.linalg.LinAlgError, r'Matrix is not positive definite'):
+            np.linalg.cholesky(A.cpu().numpy())
+
+    # TODO: once there is more support for complex dtypes on GPU, they shall be added to above test
+    # particularly when RuntimeError: _th_bmm_out not supported on CUDAType for ComplexFloat is fixed
+    @unittest.expectedFailure
+    @onlyCUDA
+    @skipCUDAIfNoMagma
+    @dtypes(torch.complex64, torch.complex128)
+    def test_cholesky_xfailed(self, device, dtype):
+        from torch.testing._internal.common_utils import random_hermitian_pd_matrix
+        A = random_hermitian_pd_matrix(shape, *batch, dtype=dtype, device=device)
+        expected_L = np.linalg.cholesky(A.cpu().numpy())
+        actual_L = torch.linalg.cholesky(A)
+        self.assertEqual(actual_L, expected_L)
+
+    # TODO: enable CUDA tests once
```

Comment:
Maybe would also be good to mention that once jit and CUDA complex support are added, this test should be moved to `test_autograd_and_jit`. Or if the tests in `test_autograd_and_jit` have been moved into `common_methods_invocations.py`, it should be moved there.

Jit and CUDA complex support can be done in future PRs.

Reply:
The test for `torch.cholesky`from `test/test_autograd.py` is not part of `common_methods_invocations.py:method_tests`. Maybe the reason is that it requires making the input hermitian for gradcheck to work correctly
```python
def func(x):
    x = 0.5 * (x + x.transpose(-1, -2).conj()) # Make `x` Hermitian
    return torch.linalg.cholesky(x)
```
`func(x)` is being tested not `torch.linalg.cholesky(x)` directly. So I don't know whether it can be moved later to `test_autograd_and_jit`/`common_methods_invocations.py`.

Reply:
As far as I understand, I don't think that prevents us from putting this in `common_methods_invocations.py:method_tests`. The second element of an entry in `method_tests` can be a contructing function, for instance: https://github.com/pytorch/pytorch/blob/master/torch/testing/_internal/common_methods_invocations.py#L521

So I think it would be possible to make a constructing function for `cholesky` too.